filters/auth/oauth.go

```diff
@@ -145,9 +149,35 @@ func intersect(left []string, right []string) bool {
 	return false
 }
 
+func createHttpClient(timeout time.Duration) *http.Client {
+	var client *http.Client
+	if timeout != 0 {
+		transport := &http.Transport{
+			DialContext: (&net.Dialer{
+				Timeout: timeout,
+			}).DialContext,
+		}
+
+		client = &http.Client{Transport: transport}
+
+		go func() {
```

Comment:
i wonder if this is required. If it is required, then we should make it possible to return from this goroutine.